### PR TITLE
Chore/puppeteer

### PR DESCRIPTION
**Pull request by ryami333**
Adds integration tests for the WAI-ARIA spec using `puppeteer` (a `chromedriver` abstraction). Little bit of a re-organisation of the 'demo' config so that it can be shared with the 'integration' implementation too. Couple of missing tests, but they are to be expanded upon as we go. 

Basically a copy and paste of [the spec](https://www.w3.org/TR/wai-aria-practices-1.1/#accordion), enumerated into test assertions.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 module.exports = {
-	testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.[jt]sx?$',
```
Comment (ryami333) [inline jest.config.js]: Just Prettier doing its thing here

File: jest.config.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
-	setupTestFrameworkScriptFile: './src/setupTestFramework.js',
-	testPathIgnorePatterns: ['/node_modules/', '/dist/'],
-	moduleFileExtensions: ['js', 'jsx', 'ts', 'tsx'],
-	transform: {
-		'^.+\\.(js|ts)x?$': 'babel-jest',
-	},
+    testRegex: '(/__tests__/.*|(\\.|/)(test|spec))\\.[jt]sx?$',
+    setupFiles: ['./src/setupTests.js'],
+    setupTestFrameworkScriptFile: './src/setupTestFramework.js',
+    testPathIgnorePatterns: ['/node_modules/', '/dist/', '/integration/'],
```
Comment (ryami333) [inline jest.config.js]: The only meaningful change here is on this line. Added 'integration' to ignores.

File: .gitignore [hunk 3]
```diff
@@ -76,9 +76,8 @@ bower_components
 # Generated files
 # -------------------------------------------------
 dist
-pages
```
Comment (ryami333) [inline .gitignore]: Changed the build location of the demo to 'demo/dist'

File: demo/src/index.html [hunk 4]
```diff
@@ -9,8 +9,7 @@
             name="description"
             content="Accessible Accordion component for React"
         />
-        <link rel="stylesheet" href="main.css" />
```
Comment (ryami333) [inline demo/src/index.html]: Now imported directly in `index.tsx`

File: integration/wai-aria.spec.js [hunk 5]
```diff
@@ -0,0 +1,297 @@
+import path from 'path';
+import { spawnSync } from 'child_process';
```
Comment (catepalmer) [inline integration/wai-aria.spec.js]: `spawnSync` doesn't seem to be being used at this point, so just wondering if it's necessary to import it?

Comment (ryami333) [reply integration/wai-aria.spec.js]: Good catch!

**Comment by ryami333** [comment]
Thanks :) Yeah, though I don't think these two will pass once they're written:

> Each accordion header button is wrapped in an element with role heading that has a value set for aria-level that is appropriate for the information architecture of the page.

> If the accordion panel associated with an accordion header is visible, and if the accordion does not permit the panel to be collapsed, the header button element has aria-disabled set to true.

That reveals a couple of holes in our implementation, which I'll raise issues for!

**Comment by catepalmer** [comment]
Yep good point, which I guess is why explicitly testing against the wai-aria spec is such an important thing to be doing!

**Pull request merged**